identity-service/src/routes/healthCheck.js

```diff
@@ -76,29 +95,70 @@ module.exports = function (app) {
       // If the max number of transactions have been evaluated, break out
       if (txCounter > maxTransactions) break
       let block = await web3.eth.getBlock(i, true)
+      if (!minBlockTime || block.timestamp < minBlockTime) minBlockTime = block.timestamp
       if (block && block.transactions.length) {
         for (const tx of block.transactions) {
           // If transaction is from audius account, determine success or fail status
           if (RELAY_HEALTH_ACCOUNT === tx.from) {
             const txHash = tx.hash
             const resp = await web3.eth.getTransactionReceipt(txHash)
             txCounter++
-            if (!resp.status) failureTxHashes.push(txHash)
+
+            // tx failed
+            if (!resp.status) {
+              const senderAddress = await redis.hget('txHasToSenderAddress', txHash)
+              if (senderAddress) {
+                if (!failureTxs[senderAddress]) failureTxs[senderAddress] = [txHash]
+                else failureTxs[senderAddress].push(txHash)
+              } else failureTxs['unknown'].push(txHash)
+            }
           }
         }
       }
     }
 
+    let isError = false
+    if (Object.keys(failureTxs).length >= RELAY_HEALTH_MIN_NUM_USERS &&
+      flatten(Object.values(failureTxs)).length > maxErrors) isError = true
+
+    // delete old entries from set in redis
+    const epochOneHourAgo = Math.floor(Date.now() / 1000) - 3600
+    await redis.zremrangebyscore('relayTxAttempts', '-inf', epochOneHourAgo)
+    await redis.zremrangebyscore('relayTxFailures', '-inf', epochOneHourAgo)
+    await redis.zremrangebyscore('relayTxSuccesses', '-inf', epochOneHourAgo)
+
+    // check if there have been any attempts in the time window that we processed the block health check
+    const attemptedTxsInRedis = await redis.zrangebyscore('relayTxAttempts', minBlockTime, '+inf')
```

Comment:
given that you iterate through range of blocks from startBlock to endBlock, and minBlockTime is the timestamp of the oldest block before startBlock or txCounter is hit, shouldn't this zrangebyscore be from minBlockTime to endBlock.timestamp?

Reply:
good point. i'm going to default end block to current block to avoid this issue. also removes cache availability problem for blocks older than 1 hour, since those would have been expired